src/App.tsx

```diff
@@ -15,36 +17,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  SORT_GOODS_ALPHABET = 'Alphabet',
+  SORT_GOODS_LENGTH = 'Length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortFild: SortType | string,
+  reversed: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortFild) {
+      case SortType.SORT_GOODS_ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.SORT_GOODS_LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortFild, setSortFild] = useState('');
```

Comment:
You can use this `default` value in `useState()`:

`const [sortFild, setSortFild] = useState(SortType.DEFAULT);`